crates/cxx-qt/src/lib.rs

```diff
@@ -50,3 +50,118 @@ pub trait CxxQtType {
 pub trait Locking {
     // empty
 }
+
+/// This trait can be implemented inside the [`#[cxx_qt::bridge]`](bridge) on any [CxxQtType] to define a
+/// custom constructor in C++ for the QObject.
+///
+/// The `Arguments` must be a tuple of CXX types that will be the arguments to the constructor in C++.
+///
+/// Note that implementing this trait will stop Cxx-Qt from generating a default constructor.
+///
+/// # Minimal Example
+/// ```no_run
+/// # #[macro_use] extern crate cxx;
+/// #[cxx_qt::bridge]
+/// mod ffi {
+///     // Struct without `Default` implementation
+///     #[cxx_qt::qobject]
+///     pub struct MyStruct {
+///         integer: i32,
+///         string: String
```

Comment:
could always bring in `cxx_qt_lib::QString`

Reply:
I was trying to avoid that to keep the code in the example minimal and not clutter it with distracting imports.
And hiding the import might have users confused why the example doesn't compile if they copy it into their own codebase.